diff -urN streaming-commons-0.1.19/System/Win32File.hsc streaming-commons-0.1.19.patched/System/Win32File.hsc
--- streaming-commons-0.1.19/System/Win32File.hsc	2015-11-24 18:15:28.000000000 +0800
+++ streaming-commons-0.1.19.patched/System/Win32File.hsc	2018-03-14 21:16:55.000000000 +0800
@@ -31,8 +31,8 @@
 
 
 #include <fcntl.h>
-#include <Share.h>
-#include <SYS/Stat.h>
+#include <share.h>
+#include <sys/stat.h>
 #include <errno.h>
 
 newtype OFlag = OFlag CInt
