--- bifunctors-5.5.2/bifunctors.cabal	2018-02-07 02:24:49.000000000 +0800
+++ bifunctors-5.5.2/bifunctors.cabal	2018-03-09 15:04:17.000000000 +0800
@@ -43,7 +43,7 @@
     base-orphans        >= 0.5.2 && < 1,
     comonad             >= 4     && < 6,
     containers          >= 0.1   && < 0.6,
-    template-haskell    >= 2.4   && < 2.13,
+    template-haskell    >= 2.4,
     th-abstraction      >= 0.2.2 && < 0.3,
     transformers        >= 0.2   && < 0.6,
     transformers-compat >= 0.5   && < 0.6
