--- system-fileio-0.3.16.3/lib/Filesystem.hs	2015-05-12 15:50:19.000000000 +0800
+++ system-fileio-0.3.16.3/lib/Filesystem.hs	2018-03-04 16:41:24.000000000 +0800
@@ -170,7 +170,7 @@
 #ifdef CABAL_OS_WINDOWS
 	let old' = encodeString old in
 	let new' = encodeString new in
-	Win32.moveFileEx old' new' Win32.mOVEFILE_REPLACE_EXISTING
+	Win32.moveFileEx old' (Just new') Win32.mOVEFILE_REPLACE_EXISTING
 #else
 	withFilePath old $ \old' ->
 	withFilePath new $ \new' ->
