diff -ru x509-1.7.2.orig/Data/X509/DistinguishedName.hs x509-1.7.2/Data/X509/DistinguishedName.hs
--- x509-1.7.2.orig/Data/X509/DistinguishedName.hs	2017-06-26 18:12:09.000000000 +0200
+++ x509-1.7.2/Data/X509/DistinguishedName.hs	2017-09-15 12:17:10.135154154 +0200
@@ -17,6 +17,7 @@
 
 import Control.Applicative
 import Data.Monoid
+import Data.Semigroup
 import Data.ASN1.Types
 import Data.X509.Internal
 
@@ -49,9 +50,12 @@
 newtype DistinguishedNameInner = DistinguishedNameInner DistinguishedName
     deriving (Show,Eq)
 
+instance Semigroup DistinguishedName where
+    (DistinguishedName l1) <> (DistinguishedName l2) = DistinguishedName (l1++l2)
+
 instance Monoid DistinguishedName where
     mempty  = DistinguishedName []
-    mappend (DistinguishedName l1) (DistinguishedName l2) = DistinguishedName (l1++l2)
+    mappend = (<>)
 
 instance ASN1Object DistinguishedName where
     toASN1 dn = \xs -> encodeDN dn ++ xs
Only in x509-1.7.2/Data/X509: DistinguishedName.hs~
